Mathlib/Data/Seq/Seq.lean

```diff
@@ -1245,6 +1260,52 @@ theorem fold_head (init : β) (f : β → α → β) (s : Seq α) :
 
 end Fold
 
+section Update
+
+variable {x hd : α} {s tl : Seq α} {n m : ℕ} {f : α → α}
+
+@[simp]
+theorem update_nil : update nil n f = nil := by
+  ext1 m
+  simp [update, Function.update]
```

Comment:
It's weird that you need to both do ext and unfold `update`. That's a sign that you're missing a projection lemma. I assume the missing lemma is something like
```
@[simp] lemma get?_update (s : Seq α) (n : ℕ) (f : α → α) (m : ℕ) :
    (s.update n f).get? m = if m = n then (s.get? m).map f else s.get? m
```